saleor/graphql/shop/types.py

```diff
@@ -222,16 +227,16 @@ def resolve_available_shipping_methods(_, info, channel, address=None):
         return resolve_available_shipping_methods(info, channel, address)
 
     @staticmethod
-    def resolve_countries(_, _info, language_code=None):
+    def resolve_countries(_, _info, language_code=None, in_shipping_zones=None):
         taxes = {vat.country_code: vat for vat in VAT.objects.all()}
-
         # DEPRECATED: translation.override will be dropped in Saleor 4.0
         with translation.override(language_code):
             return [
                 CountryDisplay(
                     code=country[0], country=country[1], vat=taxes.get(country[0])
                 )
                 for country in countries
+                if country[0] in get_countries_codes_list(in_shipping_zones)
```

Comment:
Can't we have something like this? Now we iterate over `django_countries.countries` twice- in the resolver and in the utility function while I think we can do it once.

```suggestion
                for country in get_countries_codes_list(in_shipping_zones)
```

Reply:
It will not work because translation.override works only when iteration is made on djangos `Countries` object. I can't see the way to do it with less iterations than 2 at the moment. I might miss something though.